src/sentry/runner/commands/run.py

```diff
@@ -211,10 +211,36 @@ def run_worker(**options):
 @click.option("--without-mingle", is_flag=True, default=False)
 @click.option("--without-heartbeat", is_flag=True, default=False)
 @click.option("--max-tasks-per-child", default=10000)
+@click.option("--ignore-unknown-queues", is_flag=True, default=False)
 @log_options()
 @configuration
-def worker(**options):
+def worker(ignore_unknown_queues, **options):
     """Run background worker instance and autoreload if necessary."""
+
+    from sentry.celery import app
+
+    known_queues = frozenset(c_queue.name for c_queue in app.conf.CELERY_QUEUES)
+
+    if options["queues"] is not None:
+        if not options["queues"].issubset(known_queues):
+            unknown_queues = options["queues"] - known_queues
+            message = "Following queues are not found: %s" % unknown_queues
```

Comment:
I'm not sure why you changed all of this from how I had it, but now the output is really ugly here. `%s` on a frozenset is gross.

```
>>> '%s' % frozenset(['a'])
"frozenset({'a'})"
```

Maybe something like:

```python
            message = "Following queues are not found: %s" % ",".join(sorted(list(unknown_queues)))
```

Reply:
>I'm not sure why you changed all of this from how I had it,

Because we already had sets so this logic is both more concise, efficient, and clear.

> but now the output is really ugly here. %s on a frozenset is gross.

On Py3 it was slightly better when I checked but sure, will make it pretty.

Reply:
> On Py3 it was slightly better when I checked but sure, will make it pretty.

```
Python 3.8.2 (default, Apr 29 2020, 14:27:56)
[Clang 11.0.3 (clang-1103.0.32.59)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> '%s' % frozenset(['a'])
"frozenset({'a'})"
```

🤔

Reply:
Ugh, I tried this with `set(['a'])` instead of `frozenset()`. It sure is ugly as hell.